micropip/package_index.py

```diff
@@ -174,15 +174,15 @@ def _compatible_only(
         Checking compatibility takes a bit of time, so we use a generator to avoid doing it if not needed.
         """
 
-        releases_compatible = {
-            version: cls._compatible_wheels(files, version, name=name)
+        releases_compatible_x = {
+            version: tuple(cls._compatible_wheels(files, version, name=name))
```

Comment:
I would like to keep the generator if possible, as checking compatibility of a long list of versions is redundant.